src/Illuminate/Http/Concerns/InteractsWithInput.php

```diff
@@ -224,6 +224,27 @@ public function missing($key)
         return ! $this->has($keys);
     }
 
+    /**
+     * Apply the callback if the request is missing the given input item key.
+     *
+     * @param  string  $key
+     * @param  callable  $callback
+     * @param  callable|null  $default
+     * @return $this|mixed
+     */
+    public function whenMissing($key, callable $callback, callable $default = null)
+    {
+        if ($this->missing($key)) {
+            return $callback(data_get($this->all(), $key)) ?: $this;
+        }
+
+        if ($default) {
+            return $default();
```

Comment:
`return value($default);` and eliminate callable type hint.

Reply:
The current solution is closer to how it's done in `whenHas()` and `whenFilled()` so at this stage I would prefer it like that.

Thanks for the feedback!